site/components/map/Map.tsx

```diff
@@ -261,14 +361,137 @@ const Map = ({
         setSelectAll(!selectAll);
     };
 
+    useEffect(() => {
+        setSelectAll(true);
+    }, [searchedRoutes]);
+
+    const onHover = useCallback((event: MapLayerMouseEvent) => {
+        const service = event.features && event.features[0];
+        if (service && service.properties && service.properties.serviceId) {
+            setHoverInfo({
+                longitude: event.lngLat.lng,
+                latitude: event.lngLat.lat,
+                serviceId: service && (service.properties.serviceId as number),
+            });
+        }
+    }, []);
+
+    const selectedService = (hoverInfo && hoverInfo.serviceId) || "";
+    const filter = useMemo(() => ["==", "serviceId", selectedService], [selectedService]);
+
+    const getSourcesInbound = useCallback(
+        (searchedRoutes: Partial<(Routes & { serviceId: number })[]>) =>
+            searchedRoutes.map((searchedRoute) =>
+                searchedRoute?.inbound ? (
```

Comment:
Maybe change this to searchedRoute?.inbound && searchedRoute?.serviceId so that you can just do searchedRoute.serviceId below

Reply:
done